addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1239,14 +1240,26 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
             </we-select>
         `)[0]);
         const recursiveFindCircular = (el) => {
-            if (el.dataset.visibilityDependency === this._getFieldName()) {
-                return true;
-            }
-            const dependencyInputEl = this._getDependencyEl(el);
-            if (!dependencyInputEl) {
-                return false;
+            const dependentFieldName = this._getFieldName(el);
+            // Get all the fields that have the same label than the input.
+            let fieldEls = Array.from(this.formEl
+                .querySelectorAll(`.s_website_form_input[name="${dependentFieldName}"]`))
+                .map((el) => el.closest(".s_website_form_field"));
+            // Filter the array to remove the duplicate fields. This could
+            // happen if the field has multiple inputs ("Multiple Checkboxes" for
+            // example.)
+            fieldEls = [...new Set(fieldEls)];
+            const fieldName = this._getFieldName();
+            for (const fieldEl of fieldEls) {
+                if (fieldEl.dataset.visibilityDependency === fieldName) {
```

Comment:
[Set can be iterated on](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/@@iterator), so there's no need to convert it to an array, you can just
```js
fieldEls = new Set(fieldEls);
for (const fieldEl of fieldEls) {
    ...
}
```